api/src/test/java/org/pmiops/workbench/api/ClusterControllerTest.java

```diff
@@ -234,6 +249,27 @@ public void testListClusters() throws Exception {
         .isEqualTo(testCluster);
   }
 
+  @Test
+  public void testDeleteClustersInProject() throws Exception {
```

Comment:
The testing pattern here is a bit awkward. What I might have expected to see in a controller-level test would be to make sure all inbound parameters are correctly mapped to the service-level call, e.g.

1. Set up any required mocks, e.g. when(notebookService).listClustersByProjectAsAdmin(...)
2. Call the controller method.
3. Verify that notebookService.deleteClusterAsAdmin was called with BILLING_PROJECT_ID
4. Verify that clusterAuditor.fireDeleteClustersInProject is called with BILLING_PROJECT_ID

Reply:
Its kinda tricky because all the notebookService stuff is calling to Leo. I will update though.

Reply:
I would also like to see these test cases:
* `clusterNamesToDelete.getClustersToDelete()` is null
* `clusterNamesToDelete.getClustersToDelete()` is an empty list
* delete some but not all clusters
* request to delete cluster(s) which exist but are in a different project
* returned cluster status is DELETED (I think there's a bug here)
* returned cluster status is in a non-acceptable state

Reply:
Talked offline, we are not adding the last two of these test cases because: 

- With DELETED, we don't want to receive those clusters, because we reuse names, and thus there could be many clusters with the same name in deleted.
- We could do the non-acceptable state, but it is just testing logging, and is going to be incredibly messy to maintain that test. (And is sorta testing 'what happens if Leo fails to follow contract')